File: app/src/main/java/com/grupa1/teleman/networking/JdbcConnection.java
package com.grupa1.teleman.networking;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.ResultSet;
import java.util.Properties;

import com.mysql.jdbc.Driver;

public class JdbcConnection {
    private Connection connection;

    public JdbcConnection(String IP, String PORT, String database, String username, String password){
        try {
            Class.forName("com.mysql.jdbc.Driver");
            Properties props = new Properties();
            props.setProperty("user", username);
            props.setProperty("password", password);
            //connection =  DriverManager.getConnection("jdbc:mysql://"+IP+":"+PORT+"/"+database, props);
            connection = DriverManager.getConnection("jdbc:mariadb://"+IP+":"+PORT+"/"+database, props);
            int x=0;
        } catch (Exception ex){
            ex.printStackTrace();
        }
    }

    public ResultSet executeQuery(String query){
        try {
            ResultSet result = connection.createStatement().executeQuery(query);
            if(result.isBeforeFirst()) return result;
        } catch (Exception ex){
            ex.printStackTrace();
        }
        return null;
    }

    public void closeConnection(){
        if(connection!=null){
            try {
                connection.close();
            } catch (Exception ex){
                ex.printStackTrace();
            }
        }
    }

    public boolean isConnectionOpenned(){
        if(connection!= null){
            try{
                if(!connection.isClosed()){
                    return true;
                }
            } catch (Exception ex){
                ex.printStackTrace();
            }
        }
        return false;
    }
}

File: app/src/main/java/com/grupa1/teleman/networking/JdbcConnection.kt
package com.grupa1.teleman.networking

import android.widget.Toast
import com.grupa1.teleman.MainActivity
import com.grupa1.teleman.files.ConnectionConfig
import org.jetbrains.anko.*
import org.jetbrains.anko.custom.async
import java.sql.Connection
import java.sql.DriverManager
import java.sql.ResultSet
import java.sql.SQLException

open class JdbcConnection(IP: String, PORT: String, database: String, username: String, password: String) {
    var connection: Connection? = null

    constructor(connCfg: ConnectionConfig) : this(
            connCfg.databaseAddress,
            connCfg.databasePort,
            connCfg.databaseName,
            connCfg.databaseUsername,
            connCfg.databasePassword)

    init {
        try {
            Class.forName("org.drizzle.jdbc.DrizzleDriver").newInstance()
            val connStr = "jdbc:mysql:thin://$IP:$PORT/$database"
            if (connection == null) {
                val one = doAsyncResult{ DriverManager.getConnection(connStr, username, password) }
                connection = one.get()
                val x=0
            }
        } catch (ex: SQLException) {
            System.out.println("SQLException: " + ex.message)
            System.out.println("SQLState: " + ex.sqlState)
            System.out.println("VendorError: " + ex.errorCode)
            System.out.println("")
        } catch (ex2: java.lang.Exception){
            ex2.printStackTrace()
        }

    }


    val isConnectionOpenned: Boolean
        get() {
            if (connection != null) {
                try {
                    if (connection!!.isClosed) {
                        return true
                    }
                } catch (ex: Exception) {
                    ex.printStackTrace()
                }

            }
            return false
        }

    fun executeQuery(query: String): ResultSet? {
        try {
            val task = doAsyncResult { connection!!.createStatement().executeQuery(query) }
            val result = task.get()
            if (result.isBeforeFirst)
                return result
        } catch (ex: Exception) {
            ex.printStackTrace()
        }

        return null
    }

    fun closeConnection() {
        if (connection != null) {
            try {
                connection!!.close()
            } catch (ex: Exception) {
                ex.printStackTrace()
            }

        }
    }
}
object JdbcConnectionExecution{
    fun bar(): ResultSet?{

        return null
    }
}